apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.cs

```diff
@@ -102,22 +102,28 @@ public static UrlSigner FromServiceAccountCredential(ServiceAccountCredential cr
         /// <summary>
         /// Creates a new <see cref="UrlSigner"/> instance for a custom blob signer.
         /// </summary>
-        /// <remarks>
-        /// This method is typically used when a service account credential file isn't available, either
-        /// for testing or to use the IAM service's blob signing capabilities.
-        /// </remarks>
         /// <param name="signer">The blob signer to use. Must not be null.</param>
         /// <returns>A new <see cref="UrlSigner"/> using the specified blob signer.</returns>
-        public static UrlSigner FromBlobSigner(IBlobSigner signer)
-        {
-            GaxPreconditions.CheckNotNull(signer, nameof(signer));
-            return new UrlSigner(signer, SystemClock.Instance);
-        }
+        public static UrlSigner FromBlobSigner(IBlobSigner signer) =>
+            new UrlSigner(new BlobSignerFactory(signer), SystemClock.Instance);
+
+        /// <summary>
+        /// Creates a new <see cref="UrlSigner"/> instance for a custom blob signer obtained
+        /// from the given blob signer factory.
+        /// Note the factory will be only called once, and the returned blob signer will be reused
+        /// for subsequent signing operations with this URL signer.
+        /// </summary>
+        /// <param name="signerAsyncFactory">
+        /// The blob signer async factory to use. Must not be null.
+        /// Must not return a null blob signer</param>
```

Comment:
For some reason I can't leave a suggestion, but: "When called, must not return a null reference or a task which completes with a null result."